tests/components/honeywell/test_climate.py

```diff
@@ -0,0 +1,1025 @@
+"""Test the Whirlpool Sixth Sense climate domain."""
+import datetime
+from unittest.mock import MagicMock
+
+import aiosomecomfort
+
+from homeassistant.components.climate import (
+    ATTR_AUX_HEAT,
+    ATTR_CURRENT_TEMPERATURE,
+    ATTR_FAN_MODE,
+    ATTR_HVAC_MODE,
+    ATTR_HVAC_MODES,
+    ATTR_MAX_TEMP,
+    ATTR_MIN_TEMP,
+    ATTR_PRESET_MODE,
+    ATTR_PRESET_MODES,
+    ATTR_TARGET_TEMP_HIGH,
+    ATTR_TARGET_TEMP_LOW,
+    DOMAIN as CLIMATE_DOMAIN,
+    FAN_AUTO,
+    FAN_DIFFUSE,
+    FAN_ON,
+    PRESET_AWAY,
+    PRESET_NONE,
+    SERVICE_SET_AUX_HEAT,
+    SERVICE_SET_FAN_MODE,
+    SERVICE_SET_HVAC_MODE,
+    SERVICE_SET_PRESET_MODE,
+    SERVICE_SET_TEMPERATURE,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.components.honeywell.climate import SCAN_INTERVAL
+from homeassistant.const import (
+    ATTR_ENTITY_ID,
+    ATTR_FRIENDLY_NAME,
+    ATTR_SUPPORTED_FEATURES,
+    ATTR_TEMPERATURE,
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+from homeassistant.util.dt import utcnow
+
+from . import init_integration, reset_mock
+
+from tests.common import async_fire_time_changed
+
+FAN_ACTION = "fan_action"
+PRESET_HOLD = "Hold"
+
+
+async def test_no_thermostats(
+    hass: HomeAssistant, device: MagicMock, config_entry: MagicMock
+) -> None:
+    """Test the setup of the climate entities when there are no appliances available."""
+    device._data = {}
+    await init_integration(hass, config_entry)
+    assert len(hass.states.async_all()) == 0
+
+
+async def test_static_attributes(
+    hass: HomeAssistant, device: MagicMock, config_entry: MagicMock
+) -> None:
```

Comment:
You may want to do a snapshot test of the entity registry entry and state instead: https://developers.home-assistant.io/blog/2023/02/20/snapshot-testing/

Reply:
That's a nice way to do it :) Wish it was there when I wrote the tests originally.